src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,165 @@
+
+        if (split.length == 1 || split[1].isBlank()) {
+            return new IncorrectCommand("Please enter a valid task.");
+        }
+
+        if (split[0].equals("deadline")) {
+
+            if (!split[1].contains(" /by ")) {
+                return new IncorrectCommand("Please enter a /by command to depict the deadline.");
+            }
+
+            String[] task = split[1].split(" /by ", 2);
+
+            if (task.length <= 1 || task[1].isBlank() || task[0].isBlank()) {
+                return new IncorrectCommand("Please enter a valid task and/or deadline.");
+            }
+
+            try {
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yy HHmm");
+                return new AddCommand(task[0], LocalDateTime.parse(task[1], formatter));
+            } catch (DateTimeParseException e) {
+                return new IncorrectCommand("Please enter the date & time in a valid format! (DD/MM/YY HHMM)");
+            }
+
+        } else if (split[0].equals("event")) {
+
+            if (!split[1].contains(" /from ")) {
+                return new IncorrectCommand("Please enter a /from command to depict the deadline.");
+            }
+
+            String[] task = split[1].split(" /from ", 2);
+
+            // Check if task entered is empty
+            if (task.length <= 1 || task[1].isBlank() || task[0].isBlank()) {
+                return new IncorrectCommand("Please enter a valid task.");
+            }
+
+            // Check if /to is present
+            if (!task[1].contains(" /to ")) {
+                return new IncorrectCommand("There is no /to command present. Please try again.");
+            }
+
+            String[] to = task[1].split(" /to ", 2);
+
+            if (to.length <= 1 || to[1].isBlank() || to[0].isBlank()) {
+                return new IncorrectCommand("Please enter valid to & from dates");
+            }
+
+            try {
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yy HHmm");
+                LocalDateTime from = LocalDateTime.parse(to[0], formatter);
+                LocalDateTime till = LocalDateTime.parse(to[1], formatter);
+                if (from.isAfter(till) || from.isEqual(till)) {
+                    return new IncorrectCommand("Please ensure that the date range is valid!");
+                }
+                return new AddCommand(task[0], from, till);
+            } catch (DateTimeParseException e) {
+                return new IncorrectCommand("Please enter the date & time in a valid format! (DD/MM/YY HHMM)");
+            }
+
+        } else {
+            return new AddCommand(split[1]);
+        }
+    }
+
+    private static Command validateCommand(String[] split, boolean flag) {
+        if (split.length != 1) {
+            return new IncorrectCommand("Please enter a valid command!");
+        }
+        return flag ? new ExitCommand() : new ListCommand();
```

Comment:
I'm not a fan of using ```flag``` to choose between two very different commands. I think it would be better to make this more intuitive.